pkg/server/registry/apigroups/acorn/containers/translator.go

```diff
@@ -30,8 +30,9 @@ type Translator struct {
 
 func (t *Translator) FromPublicName(ctx context.Context, namespace, name string) (string, string, error) {
 	for {
-		prefix, suffix, ok := strings.Cut(name, ".")
-		if !ok {
+		prefix, suffix, _ := strings.Cut(name, ".")
+		// stop when name = <pod name> + "." + <container name>
```

Comment:
Hesitant on this function, especially without seeing any unit tests covering the different ways the for loop can execute. Other then this, lgtm. Nice to see tests written and all passing!

Reply:
I have added the unit tests